### DISCUSSION TITLE
How do I customize cockpit GUI ?

### DISCUSSION THREAD

**Discussion by MRajibH** [cockpit-project/cockpit#20434] [Q&A] [answered]
I have tried the method given in the link : https://github.com/cockpit-project/cockpit/blob/main/HACKING.md  but it is not working . My changes are not taking effect. Is there  any simpler method available to customize ? FYI,  I am running Cockpit on my debian 12 server.

**Comment by martinpitt** [comment]
Please give some more details what exactly you tried to do. Did you follow [Working on Cockpit's session pages](https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#working-on-cockpits-session-pages) or some other section? Which page did you modify locally? What does `cockpit-bridge --packages` show?

**Reply by MRajibH** [reply]
root@rajib:~/cockpit# cockpit-bridge --packages
```
apps                 Applications                             /root/.local/share/cockpit/apps
base1                                                         /root/.local/share/cockpit/base1
dashboard                                                     /usr/share/cockpit/dashboard
kdump                Kernel dump                              /root/.local/share/cockpit/kdump
machines             Virtual machines                         /usr/share/cockpit/machines
metrics                                                       /root/.local/share/cockpit/metrics
network              Networking                               /root/.local/share/cockpit/networkmanager
pcp                                                           /usr/share/cockpit/pcp
performance                                                   /usr/share/cockpit/tuned
playground           Development                              /root/.local/share/cockpit/playground
podman               Podman containers                        /usr/share/cockpit/podman
selinux              SELinux                                  /root/.local/share/cockpit/selinux
shell                                                         /root/.local/share/cockpit/shell
sosreport            Diagnostic reports                       /root/.local/share/cockpit/sosreport
ssh                                                           /usr/share/cockpit/ssh
static                                                        /root/.local/share/cockpit/static
storage              Storage                                  /root/.local/share/cockpit/storaged
system               Overview, Logs, Services, Terminal       /root/.local/share/cockpit/systemd
updates              Software updates                         /root/.local/share/cockpit/packagekit
users                Accounts  
```

I tried to change the files from different folder` /root/cockpit/dists` where I pulled code from github  but changes are not working. can you please tell if i want change overview sidenav  name or hide software updates sidenav of cockpit ,  what files I exactly need to change and how ?

I also wanted to add an ISO file upload option which is missing in virtual machine section. Is that possible?. Same goes for podman container where currently have no option to upload docker image from local pc. 
 Thanks

**Reply by martinpitt** [reply, marked as answer]
Ah, *please* don't do that -- develop cockpit (and in general, *everything*) as user, not root. `/root/.local` is not a thing, that's /usr/share/local/. To clean up, do `rm -r /root/.local`.
